### ISSUE TITLE
Make site export location-agnostic

### ISSUE THREAD

**Issue by j9t** [tweetback/tweetback#13]
First brief glance, may also prepare a PR if I get to this:

The site export works with root-relative URLs (e.g., `<link rel="stylesheet" href="/assets/chartist.min.css">`). This means these references don’t work if the export is moved to a subfolder, or if tested on the file system.

The most versatile solution is probably to make these references site-relative (e.g., `<link rel="stylesheet" href="assets/chartist.min.css">`, or `<link rel="stylesheet" href="../assets/style.css">` for tweet pages).

Alternatively, a base path or URL could help.

**Comment by idleberg** [comment]
The 11ty CLI has a parameter for a path prefix. This allows you, for instance, to publish the page to GitHub Pages eithout 404s.

`npx @11ty/eleventy --quiet --pathprefix 'my-repository-name'`

**Comment by zachleat** [comment]
This _is_ working with `--pathprefix` from the command line and `tweetback` is now using the Eleventy `<base>` plugin to ship this feature.

It was fixed in https://github.com/tweetback/tweetback/commit/9fbbb2bf8191bbb6d373cd2a5e6ef38caef4968b

NOTE: There was a regression with `pathPrefix` returned in configuration file that will be fixed in Eleventy core v2.0.0-canary.19: https://github.com/11ty/eleventy/pull/2592

**Comment by zachleat** [comment]
You can read more about the `<base>` plugin here: https://www.11ty.dev/docs/plugins/html-base/

**Comment by zachleat** [comment]
Hmm, I just tested this and it seems to work for me @j9t:

```
npx @11ty/eleventy --quiet --pathprefix=my-folder
```

and then I opened up `_site/index.html` which had the following:

```
			<link rel="stylesheet" href="/my-folder/assets/chartist.min.css">
			<link rel="stylesheet" href="/my-folder/assets/chart.css">
			<script src="/my-folder/assets/chartist.min.js"></script>
			<script src="/my-folder/assets/chart.js"></script>
```

**Comment by j9t** [comment]
Thanks @zachleat. Tested with the current version of tweetback, as a CLI parameter, and now it does work for me, too. Strange. If I find time I’ll check again—originally I had tried with `pathPrefix` in eleventy.config.js. Thanks for following up.

**Comment by zachleat** [comment]
ah yes as noted in the above comment `pathPrefix` in config is currently broken (although already fixed for the next canary release `v2.0.0-canary.19`) https://github.com/tweetback/tweetback/issues/13#issuecomment-1325304013

**Comment by cooljeanius** [comment]
> Hmm, I just tested this and it seems to work for me @j9t:
> 
> ```
> npx @11ty/eleventy --quiet --pathprefix=my-folder
> ```
> 
> and then I opened up `_site/index.html` which had the following:
> 
> ```
> 			<link rel="stylesheet" href="/my-folder/assets/chartist.min.css">
> 			<link rel="stylesheet" href="/my-folder/assets/chart.css">
> 			<script src="/my-folder/assets/chartist.min.js"></script>
> 			<script src="/my-folder/assets/chart.js"></script>
> ```

This gets missing images and formatting to show up properly for me, but the pagefind stuff still seems to be broken, though?
